engine/crates/composition/src/subgraphs/definitions.rs

```diff
@@ -90,3 +90,17 @@ impl<'a> DefinitionWalker<'a> {
         self.walk(self.definition().subgraph_id)
     }
 }
+
+impl<'a> SubgraphWalker<'a> {
```

Comment:
Btw, why we want to split these implementations to different modules? Now it's a bit tricky to have a full vision what is implemented in `SubgraphWalker`, if it's in multiple places.

Reply:
I'm experimenting with this because it lets me keep the data in the `subgraphs::keys` module private (for example), so everything interacting with that piece of data is encapsulated in the module. Jury is still out on whether that's a better approach, but the crate is structured like that at the moment.